[UX Dashboard] Move core web vitals query from APM routes to UX plugin

justinkambic opened the pull request:
## Summary

Related to [#454](https://github.com/elastic/uptime/issues/454).

Moves CWV query/fetch logic entirely to UX plugin.

### Checklist

Delete any items that are not applicable to this PR.

- [X] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

### Risk Matrix

Delete this section if it is not applicable to this PR.

Before closing this PR, invite QA, stakeholders, and other developers to identify risks that should be tested prior to the change/feature release.

When forming the risk matrix, consider some of the following examples and how they may potentially impact the change:

| Risk                      | Probability | Severity | Mitigation/Notes        |
|---------------------------|-------------|----------|-------------------------|
| Multiple Spaces&mdash;unexpected behavior in non-default Kibana Space. | Low | High | Integration tests will verify that all features are still supported in non-default Kibana Space and when user switches between spaces. |
| Multiple nodes&mdash;Elasticsearch polling might have race conditions when multiple Kibana nodes are polling for the same tasks. | High | Low | Tasks are idempotent, so executing them multiple times will not result in logical error, but will degrade performance. To test for this case we add plenty of unit tests around this logic and document manual testing procedure. |
| Code should gracefully handle cases when feature X or plugin Y are disabled. | Medium | High | Unit tests will verify that any feature flag or plugin combination still results in our service operational. |
| [See more potential risk examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx) |


### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

justinkambic committed:
Migrate service list query out of APM.

justinkambic committed:
Migrate service list query out of APM.

justinkambic committed:
Migrate service list query out of APM.

justinkambic committed:
Migrate service list query out of APM.

justinkambic committed:
Rename non-snakecase files.

justinkambic committed:
Rename non-snakecase files.

justinkambic committed:
Rename non-snakecase files.

justinkambic committed:
Rename non-snakecase files.

justinkambic committed:
Move core web vitals query out of APM routes.

justinkambic committed:
Move core web vitals query out of APM routes.

justinkambic committed:
Delete obsolete snapshot.

justinkambic committed:
Delete obsolete snapshot.

justinkambic committed:
Refactor o11y overview registration to not rely on deleted APM route.

justinkambic committed:
Refactor o11y overview registration to not rely on deleted APM route.

justinkambic committed:
Fix some types.

justinkambic committed:
Fix some types.

justinkambic committed:
Delete obsolete API test.

justinkambic committed:
Delete obsolete API test.

justinkambic committed:
Delete obsolete APM API tests.

justinkambic committed:
Delete obsolete APM API tests.

justinkambic committed:
Delete CWV test from APM API suite.

justinkambic committed:
Delete CWV test from APM API suite.

efegurkan committed:
[Enterprise Search] Remove content plugin for 8.3 release (#133738)

* Remove content plugin for 8.3 release

* Delete server side plugins

* Remove collection schema

* Fix CI

justinkambic committed:
Merge branch 'main' into 454/core-web-vitals

jgowdyelastic committed:
[ML] Tests for get groups endpoint (#133625)

* [ML] Tests for get groups endpoint

* adding extra calendar

* removing delete index pattern

mistic committed:
docs(NA): adds @kbn/plugin-discovery into ops devdocs (#133632)

* docs(NA): adds @kbn/plugin-discovery into ops devdocs

* chore(NA): update packages/kbn-plugin-discovery/README.mdx

Co-authored-by: Jonathan Budzenski <jon@budzenski.me>

Co-authored-by: Jonathan Budzenski <jon@budzenski.me>

CohenIdo committed:
[Cloud Posture] update retention policy and remove score transform configuration (#133730)

vadimkibana committed:
ui_actions_enchanced to /src (#133512)

* move ui_actions_enhanced to /src

* [CI] Auto-commit changed files from 'node scripts/precommit_hook.js --ref HEAD~1..HEAD --fix'

* [CI] Auto-commit changed files from 'node scripts/build_plugin_list_docs'

* move translations to /src

* fix typescript errors

* update config files

* update ts configs

* fix config path

* [CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

lukasolson committed:
[Data views] Implement persistable state (#132959)

* Tmp commit

* Use serializable version

* Update tests

* Add test for fields

* Use spec instead of base

* Update types

* Fix type errors

* Fix tests & types

* Update types

* Fix lint

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

qn895 committed:
[APM] Fix ML expected bounds missing data in chart, option showing when not neccessary (#132975)

* Only show expected bounds option in Overview/Transactions subpage of Services

* Bucket span

* Fix bucket span didn't show up in chart

* Change fixed interval to 60s, add elastic-charts ref

* ui settings remembering

* Address comments

* Address comments

* Fix linting

* Move data point down instead of extending range

* Change to last, add test

* Fix to use xValuesExpectedBounds

* Revert change to to add extra point as es ml already handled it

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

semd committed:
[SecuritySolution] Dashboards section title in the old nav (#133719)

* add dashboards section title to old nav

* test fixed

* snapshot rollback

janmonschke committed:
fix: make sure createThresholdTimeline works when no filters are provided (#133733)

Commit:
[Archive Migration] Kbn archiver management (#133675)

* remove kibana saved object from the esArchive

* migrate management from esArchive to kbnArchive

* migrate management from esArchive to kbnArchive

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

juliaElastic committed:
fixed tags count (#133788)

darnautov committed:
[ML ]API integration tests for `max_anomaly_score` endpoint (#133776)

* max_anomaly_score tests

* 404 test

* remove tag

* fix group name and a typo

elasticmachine commented:
Pinging @elastic/uptime (Team:uptime)

justinkambic committed:
Re-introduce required id types for `FormatType`. (#133812)

nickpeihl committed:
[Spacetime] [Maps] Customizable colors in basemaps (#131576)

* Customizable colors in basemaps

* Add operations to color selector

* Only use default color operations per style

Allowing the user to choose the operation is probably unnecessary at this step. The default operations work very well.

* Fix types

* Clean up cruft

* Update ems-client

* Fix license override

* [CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

* Fix test

* Unneccesary type assertion

* Refactor color filters to use new EMSVectorTileStyle

* lint

* [CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

* Review feedback

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

Commit:
migrate to kbnArchive (#133683)

Commit:
[Archive Migration] remove unused test/functional/fixtures/es_archiver/saved_objects_imports (#133818)

* remove unused es_archive

* remove unused es_archiver/timelion

sorenlouv reviewed (approved):
lgtm

mattkime committed:
update docs (#133798)

flash1293 committed:
fix legend inside handling (#133732)

Co-authored-by: Stratoula Kalafateli <efstratia.kalafateli@elastic.co>

shahzad31 reviewed (approved):
LGTM !!

Please add a simple e2e journey test if possible !!

flash1293 committed:
do not show edit field for record field (#133762)

Co-authored-by: Stratoula Kalafateli <efstratia.kalafateli@elastic.co>

CohenIdo committed:
[Cloud Posture] use internal user for package policy changes (#133293)

orouz committed:
[Cloud Posture] show agent name in rules page description (#133758)

miltonhultgren committed:
[Stack Monitoring] Add stale status reporting for Kibana (#132613)

* [Stack Monitoring] Add stale status reporting for Kibana (#126386)

* Fix stale message grammar and update stale indicator to use EuiBadge

* Fix i18n ids

* Remove unused i18n key

* Fix Jest tests

* Update exposeToBrowser test

* Update API integration tests

* Fix functional tests

* Fix API integration tests

* Update snapshots

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

criamico committed:
[Fleet] Prevent upgrades of agents already in updating state (#133763)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

szabosteve committed:
[DOCS] Fixes variable names in Index connector docs. (#133725)

pgayvallet committed:
Create core packages for docLinks browser/server services (#133360)

* Create core packages for docLinks browser/server services

* fix mock imports

* fix service imports

* fix server-side type imports

* fix type imports on browser side

* updating readme files

* update generated doc

* add base server mock package

* adapt core usages of coreContextMock

* fix tests / mocks

* lint

* add missing export

* fix another test import

* import from package

shahzad31 committed:
[Synthetics] Load locations only once (#133564)

orouz committed:
[Cloud Posture] remove tooltip from switch button (#133860)

flash1293 committed:
fix application of suffixes (#133780)

academo committed:
[Cases] refactor useGetCases to use react-query (#133505)

fkanout committed:
[Actionable Observability][BUG] Fix connectors doesn't appear in the Rule Details page after creating first connector (#133737)

* Move rule actions to the hook which is right call

* Update naming

* Fix when a connector is deleted from connector page  in Stack Management

* Fix tests

* Update tests and mock loadAllActions API

* Remove unused import

* Call the connectors API only when the rule has actions

shahzad31 committed:
[Synthetics] Script field,  updated validation and width (#133558)

Co-authored-by: Abdul Zahid <awahab07@yahoo.com>

breehall committed:
Upgrade EUI to v58.0.0 (#132257)

* Initial commit for EUI 57.0.0 upgrade

* Handle i18n changes

* Resolved type errors in DatePicker and Markdown Editor

* Resolve test failures in Jest Test Suite #1. Updated multiple snapshots for euiLink and euiTitle as they have been converted to Emotion

* Resolved failing tests for Jest Suite #2. Updated snapshots for euiHealth, euiAvatar, euiSpacer, euiTitle, and euiLink as they have recently been converted to Emotion

* Resolved failing tests for Jest Suite 3. Updated failing snapshots as EuiSpacer, EuiText, EuiCallout, EuiHorizontalRule, EuiTitle, and EuiLink have been converted to Emotion. Updated the i18n translation snapshots

* Upgrade EUI verion to 58.0.0

* Resolved tests failures from Jest Test Suite 4. Updated snapshots as EuiLink, EuiTitle, EuiHorizontalRule, EuiSpace, and EuiCallout have been converted to Emotion

* Resolved failing test cases for Jest Test Suite 5. Updated snapshots as EuiLoader has been converted to Emotion

* Resolved failing tests in Jest Test Suite 6. Updated snapshots as EuiSpacer, EuiHorizontalRule, Eui Callout, and EuiLink have been converted to Emotion

* Resolved type errors for EuiDatePicker component

* Resolved type error within EuiContextMenu by removing the watchedItemProps prop. It was recently deprecated in EUI PR# 5880 (https://github.com/elastic/eui/pull/5880) as is no longer needed

* Resolved type error within EuiContextMenu by removing the watchedItemProps prop. It was recently deprecated in EUI PR# 5880 (https://github.com/elastic/eui/pull/5880) as is no longer needed

* Resolved type error within EuiContextMenu by removing the watchedItemProps prop. It was recently deprecated in EUI PR# 5880 (https://github.com/elastic/eui/pull/5880) as is no longer needed

* Resolved type errors by updating the popoverPlacement prop for the EuiDatePicket component with new / valid values. A list of values were deprecated and new values were added in EUI PR #5868 (https://github.com/elastic/eui/pull/5868)

* Resolved type error within EuiTabs by removing instances of display: condensed as it is no longer a part of the Amsterdam theme via EUI PR #5868(https://github.com/elastic/eui/pull/5868)

* Remove deprecated `display` prop from EuiTabs

* Deprecate `.eui-textOverflowWrap`

* Deprecate EuiSuggestItem `labelDisplay` prop

* [EuiStepsHorizontal] Replace deprecated `isComplete`/`isSelected` with `status`

* Update last EuiStepsHorizontal `status` migration

- this one was more complex than the previous commit due to existing `status` usage and conditional steps. Some amount of logic was simplified via `completedStep`

* Resolved type error within EuiTabs by removing instances of display: condensed as it is no longer a part of the Amsterdam theme via EUI PR #5868(https://github.com/elastic/eui/pull/5868)

* Resolved failing test cases in Jest Test Suite 5. Updated snapshots as EuiTitle has been converted to Emotion

* Resovlved failing test cases in Jest Test Suite 4. Updated snapshots as EuiTitle and EuiSpacer have been converted to Emotion. Resolved failing tests for EuiLink click simlulations by esuring the test is referencing the correct element.

* Resolved failing test cases in Jest Test Suite 3. Updated snapshots as EuiLink, EuiSpacr, and EuiTItle have been converted to Emotion. Updated various test cases to ensure that the references to EuiLink are correct

* [CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

* Resolved failing test cases in Jest Test Suite 1. Updated snapshots as EUI text utilities have been converted to Emotion. Updated referenes to EuiLink to ensure test are simulating clicks on the correct elements

* Resolved failing test for Jet Test Suite 2. Updated required snapshots. Updated references to EuiLink to ensure that tests are simulating clicks on the correct elements

* [CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

* Resolved failing test cases for Jest Test Suite 5. Updated references to EuiLink to ensure tests are targeting the correct element

* [CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

* Resolved failing test cases across the Jest Test Suites. Updated required snaphots for components recently converted to Emotion. Updated test cases to ensure that tests targeting EuiLink are using the correct element.

* Resolved failing tests from multiple Jest test suites. Updated snapshots for components that have recently been converted to Emotion. Updated tests that reference the EuiLink component to ensure the correct element is being targeted

* Updated the getEuiStepsHorizontal function. Previously, this function used the .euiStepHorizontal-isSelected class (now deprecated) to determine which step was current. The function has been updated to use the status prop.

* Updated Jest integration test snapshots to account for the recent conversion of EuiLoader to Emotion

* Resolved failing tests in Jest suites 2 and 4. Updated required snapshots and references for tests using EuiLink

* Removed a console statement. Extracted a nested turnary operation into its own function.

* Rollback new turnary function and replace it with a simple if/else

* Rollback new turnary function and replace it with a simple if/else

* Rollback new turnary function and replace it with a simple if/else

* Rollback new turnary function and replace it with a simple if/else

* [CI] Auto-commit changed files from 'node scripts/precommit_hook.js --ref HEAD~1..HEAD --fix'

* Resolved failing test cases in Jest Suites 3 and 5 by updating required snapshots

* revert doc_viewer_source test and snapshot changes

* Take care of merge conflict in license_checker:

* Reverted .render() change for analytics_no_data_page.component.test.tsx. Restored snapshot

* Reverted .render() change for analytics_no_data_page.component.test.tsx. Restored snapshot

Co-authored-by: Constance Chen <constance.chen@elastic.co>
Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Greg Thompson <thompson.glowe@gmail.com>

lcawl committed:
[DOCS] Fix ML sync API example (#133585)

alisonelizabeth committed:
Migrate observability to new Kibana page template (#133021)

Commit:
Re-enable brolen ML tests (#133510)

Commit:
[8.4] [Security Solution] [Kubernetes Security] Move kubernetes link to dashboard in new security nav (#133411)

* Move kubernetes link to dashboard in new security nav

* Add kubernetes page image and description

estermv committed:
[Unified Observability] Remove core and plugins from PluginContext (#133466)

* replace core and plugins with specific dependencies

* remove core and plugins from tests

* clean unused plugins variable

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

pjhampton committed:
Add a default request timeout for Axios telemetry connections. (#133615)

* Add a default request timeout for axios telemetry connections.

* Update synthetics test.

* Update telemetry tests to include new timeout config.

* Skip test that is erroring out.

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

justinkambic committed:
Add test journey for core web vitals.

justinkambic committed:
Merge branch '454/core-web-vitals' of github.com:justinkambic/kibana into 454/core-web-vitals

justinkambic commented:
Sorry everyone for the pings! I will re-open this one.
